iceoryx_examples/small_memory/README.md

```diff
@@ -0,0 +1,76 @@
+# icehello with minimal memory usage
```

Comment:
Checked the example and it worked well thanks to the good description. Will continue reviewing.

Reply:
@JakubSosnovec To make this example visible on the website you can add a simple redirection with a title in https://github.com/eclipse-iceoryx/iceoryx/tree/master/doc/website/examples to link to this markdown file.
Additionally please add an entry in https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/examples/.pages

Output looks like this: https://iceoryx.io/v2.0.2/examples/

Reply:
What to do if a new example is created is also documented [here](https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#how-to-add-a-new-example).

Reply:
@dkroenke @mossmaurice I have now added the example documentation and links according to https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#how-to-add-a-new-example, including the asciinema recording (`sed`ing the user to `hypnotoad` screwed up the recording a bit, because of the different lengths of my username and `hypnotoad`, so I used `htoad` instead).

Please take a look :)